src/h5web/visualizations/containers/NxSpectrumContainer.tsx

```diff
@@ -1,39 +1,58 @@
 import React, { ReactElement, useState, useContext } from 'react';
 import { range } from 'lodash-es';
-import { HDF5SimpleShape } from '../../providers/models';
-import { useDatasetValue } from './hooks';
-import { assertGroup } from '../../providers/utils';
+import { HDF5SimpleShape, HDF5Dataset } from '../../providers/models';
+import { useDatasetValues } from './hooks';
+import { assertGroup, isDataset } from '../../providers/utils';
 import DimensionMapper from '../../dimension-mapper/DimensionMapper';
 import { DimensionMapping } from '../../dimension-mapper/models';
 import MappedLineVis from '../line/MappedLineVis';
 import { ProviderContext } from '../../providers/context';
-import { getSignalDataset } from '../nexus/utils';
+import {
+  getSignalDataset,
+  getLinkedEntity,
+  getAxesLabels,
+} from '../nexus/utils';
 import { VisContainerProps } from './models';
 
 function NxSpectrumContainer(props: VisContainerProps): ReactElement {
   const { entity } = props;
   assertGroup(entity);
 
   const { metadata } = useContext(ProviderContext);
-  const dataset = getSignalDataset(entity, metadata);
+  const signalDataset = getSignalDataset(entity, metadata);
 
-  if (!dataset) {
+  if (!signalDataset) {
     throw new Error('Signal dataset not found');
   }
 
-  const { dims } = dataset.shape as HDF5SimpleShape;
+  const { dims } = signalDataset.shape as HDF5SimpleShape;
   if (dims.length < 1) {
     throw new Error('Expected dataset with at least one dimension');
   }
 
+  const axesLabels =
+    getAxesLabels(entity) || Array(dims.length).fill(undefined);
+  if (dims.length !== axesLabels.length) {
+    throw new Error(
+      `Number of axes ${axesLabels.length} does not correspond to the dataset dimensions (${dims})`
+    );
+  }
+  const axesDatasets: Record<string, HDF5Dataset> = {};
+  axesLabels.forEach((a) => {
+    const dataset = a && a !== '.' && getLinkedEntity(entity, metadata, a);
+    if (dataset && isDataset(dataset)) {
+      axesDatasets[a] = dataset;
+    }
+  });
```

Comment:
You can do this with `axesLabels.reduce<Record<string, HDF5Dataset>>(..., {});`

Reply:
and you can store the datasets IDs directly in the record.

Reply:
> You can do this with `axesLabels.reduce<Record<string, HDF5Dataset>>(..., {});`

I tried this first but TS wouldn't let me. I will it give another go.

Reply:
It works ! I was forgetting the template in `reduce`.